### ISSUE TITLE
Allow Array field type

### ISSUE THREAD

**Issue by abdonrd** [strapi/strapi#761]
It would be great to have the `Array` type.

An example of my use case:

I have three models: `Document`, `Writer` & `Company`.
A document is created by a group of writers. I need to store the company in which each writer was in that document. I can't store the company in the writer because a the writers can change company.

`Document` model

```
{
  ...
  "authors" : [ 
    {
      "writerId": ObjectId("5aa1efaef2a732375cb64a80"),
      "companyId": ObjectId("5aa1f259fc33c43e5fa8a6f5")
    }
  ]
  ...
}
```

**Comment by lauriejim** [comment]
You can make a deep populate to get each authors and there company.
Then you can get compagnies there are linked to you Document.

**Comment by abdonrd** [comment]
@lauriejim sorry, but I have not understood it well. Could you show me an example?

**Comment by lauriejim** [comment]
Checkout http://mongoosejs.com/docs/populate.html#deep-populate.

But in your case, get `Document` populate `Author` and deep populate `Company`.
Then you can get all companies linked to your article.

**Comment by abdonrd** [comment]
@lauriejim But in this case of use, the `Writer` does not have in his model the `Company` for which he works; I don't want to store it because I only want to know in which `Company` the `Writer` was working when that `Document` was made. The idea is like:

`Writer`
```javascript
{
  "_id": ObjectId("5aa1efaef2a732375cb64a80"),
  "name": "Jim"
}
```

`Company`
```javascript
{
  "_id": ObjectId("5aa1f259fc33c43e5fa8a6f5"),
  "name": "Strapi Solutions"
}
```

`Document`
```javascript
{
  "_id": ObjectId("5aad4b07a1ff586b57064f02"),
  "name": "Document name",
  "datetime": ISODate("2017-12-15T00:00:00.000Z"),
  "authors" : [ 
    {
      "writerId": ObjectId("5aa1efaef2a732375cb64a80"),  // Jim
      "companyId": ObjectId("5aa1f259fc33c43e5fa8a6f5")  // Strapi Solutions
    }
  ]
}
// In this document, we know that Jim was working (december 2017) in Strapi Solutions.
// Maybe now (march 2018) Jim works in another Company. I don't care.
```

**Comment by lauriejim** [comment]
Ho okay, I understand your use case.
Actually you will have to store it in an other collection that has a relations with each of your other models.

Sadly we will not have time to work on now.

**Comment by abdonrd** [comment]
@lauriejim any recommendation about how to deal with this?

(right now the admin UI is not important for me)

**Comment by lauriejim** [comment]
Hey @abdonrd. The best way actually is to create a `Author` model which have a relation with `Company` and `Writer`. And your `Document` model has a relation with many `Authors`.

**Comment by abdonrd** [comment]
@lauriejim But in that way... I can't create the `Document` at once, right?

I need to do `POST /document`:

```javascript
{
  "name": "Document name",
  "datetime": ISODate("2017-12-15T00:00:00.000Z"),
  "authors" : [ 
    {
      "writerId": ObjectId("5aa1efaef2a732375cb64a80"),  // Jim
      "companyId": ObjectId("5aa1f259fc33c43e5fa8a6f5")  // Strapi Solutions
    }
  ]
}
```

And not:

1. `POST /documentauthor`:

```javascript
{
  "writerId": ObjectId("5aa1efaef2a732375cb64a80"),  // Jim
  "companyId": ObjectId("5aa1f259fc33c43e5fa8a6f5")  // Strapi Solutions
}
```

2. `POST /document`:

```javascript
{
  "name": "Document name",
  "datetime": ISODate("2017-12-15T00:00:00.000Z"),
  "authors" : [ 
    ObjectId("id-from-the-step-one")
  ]
}
```

**Comment by lauriejim** [comment]
Yes @abdonrd that's it. Actually I have no idea of how did it in one request.
This relational data structure has not be imagined and implemented in Strapi.

**Comment by MurzNN** [comment]
Our company need this feature too! Create new models for each muti-value field is too expensive!
For example, simple popular `Contact` model already need this, for store object like:
```
{
  "name": "Ivan Petrov",
  "contactPhones : [ 
    {
      "type": "Home",
      "value: "+12345678900,
    },
    {
      "type": "Work",
      "value: "+12345678901,
    },
    {
      "type": "Work2",
      "value: "+12345678902,
    },
  ]
}
```
Without this feature, we must implement new model "contactNumber" and make 4 queries to API for load the contact model?

**Comment by lauriejim** [comment]
Hello @MurzNN.
Now the `group` field type is available for this kind of need.

**Comment by lauriejim** [comment]
There is no other solution for now. We are thinking about adding the `repeatable` option for other fields that just `group`.
But it will not be now.
So yes you will have to use a group for repeatable numbers.

**Comment by derrickmehaffy** [comment]
The manyWay is also useful for this @MurzNN
